go/teams/loader2.go

```diff
@@ -256,22 +256,33 @@ func (l *TeamLoader) verifyLink(ctx context.Context,
 	}
 
 	minRole := link.outerLink.LinkType.RequiresAtLeastRole()
+	linkType := link.outerLink.LinkType
 	// Note: If minRole is OWNER it will be treated as ADMIN here (weaker check).
 	if !ShouldSuppressLogging(ctx) {
-		l.G().Log.CDebugf(ctx, "verifyLink minRole:%v", minRole)
+		l.G().Log.CDebugf(ctx, "verifyLink: %v minRole:%v", linkType, minRole)
 	}
 
 	switch minRole {
 	case keybase1.TeamRole_NONE:
 		// Anyone can make this link. These didn't exist at the time.
 		return &signer, nil
+	case keybase1.TeamRole_BOT:
+		err = l.verifyExplicitPermission(ctx, state, link, signerUV, keybase1.TeamRole_BOT)
+		if err == nil {
+			return &signer, err
+		}
+		if !ShouldSuppressLogging(ctx) {
+			l.G().Log.CDebugf(ctx, "verifyLink: %v not a %v: %v", linkType, keybase1.TeamRole_BOT, err)
+		}
+		// Fall through to a higher role check
+		fallthrough
```

Comment:
This makes `RequiresAtLeastRole -> bot` actually mean `RequiresAtLeastRole -> reader`. Which doesn't make much sense to me. If bots can sign links this shouldn't fall through. If they can't, `RequiresAtLeastRole` shouldn't return `bot`.

Reply:
i thought the `fallthrough` was really acting as a goto to the implicit admin case. `verifyExplicitPermission` is (confusingly) checking `atOrAbove` for the role so failing a lower permission check would imply failing all of the subsequent cases except if you're an offchain implicit admin https://github.com/keybase/client/pull/18206/files#diff-53390555ab6dc886c9994361396f9627R300

Reply:
Yup my bad. fallthrough looks ok